topologytest/hlv_test.go

```diff
@@ -176,6 +346,49 @@ func TestHLVResurrectDocumentSingleActor(t *testing.T) {
 	}
 }
 
+func TestHLVResurrectDocumentMultiActor(t *testing.T) {
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyImport, base.KeyVV)
+	for _, tc := range getMultiActorTestCases() {
+		t.Run(tc.description(), func(t *testing.T) {
+			if strings.Contains(tc.description(), "CBL") {
+				t.Skip("Skipping Couchbase Lite test, does not know how to push a deletion yet CBG-4257")
+			}
+
+			t.Skip("skipped resurrection test, intermittent failures CBG-4372")
+
+			peers, _ := setupTests(t, tc.topology, "")
+
+			var docVersionList []BodyAndVersion
+			peerList := tc.PeerNames()
+
+			for _, peerName := range peerList {
+				tc.originalActor = peerName
+				docID := tc.docID(tc.getOriginalActor())
+				body1 := []byte(fmt.Sprintf(`{"peer": "%s", "topology": "%s","writePeer": "%s"}`, docID, tc.description(), peerName))
+				createVersion := peers[peerName].CreateDocument(tc.collectionName(), docID, body1)
+				t.Logf("createVersion: %+v for docID: %s", createVersion, tc.docID(tc.originalActor))
+				waitForVersionAndBody(t, tc, peers, createVersion, body1)
+
+				deleteVersion := peers[peerName].DeleteDocument(tc.collectionName(), docID)
+				t.Logf("createVersion: %+v, deleteVersion: %+v", createVersion, deleteVersion)
+				t.Logf("waiting for document %s deletion on all peers", docID)
+				waitForDeletion(t, tc, peers)
+				// recreate doc and assert it arrives at all peers
+				resBody := []byte(fmt.Sprintf(`{"peer": "%s", "topology": "%s", "write": "resurrection on peer %s"}`, tc.originalActor, tc.description(), peerName))
+				updateVersion := peers[peerName].WriteDocument(tc.collectionName(), docID, resBody)
+				//waitForVersionAndBodyOnNonActivePeers(t, tc, peers, updateVersion, resBody, peerName)
+				docVersionList = append(docVersionList, BodyAndVersion{body: resBody, docMeta: updateVersion, updatePeer: peerName})
+			}
+
+			for i, updatePeer := range peerList {
+				tc.originalActor = updatePeer
+				docVersion := docVersionList[i]
+				waitForVersionAndBodyOnNonActivePeers(t, tc, peers, docVersion.docMeta, docVersion.body, docVersion.updatePeer)
```

Comment:
You can have this function take docID so that topology.originalActor is not modified.